### PR TITLE
fix(gql-api): restore CatchGatewayError decorator

### PR DESCRIPTION
**Pull request by chenba**
Because:
 - new settings was receiving JSON as error messages

This commit:
 - restore the decorator that was used to catch and re-throw errors
   prior to the NestJS conversion

## Issue that this pull request solves

Closes: #7128

### CODE REVIEW

**Comment by chenba** [comment]
@bbangert I don't know why CatchGatewayError was removed.  Let me know if this is not the correct approach.

File: packages/fxa-graphql-api/src/gql/lib/error.ts [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  target: any,
+  propertyKey: string,
+  descriptor: PropertyDescriptor
+) {
+  const originalMethod = descriptor.value;
+  descriptor.value = async function (...args: any[]) {
+    try {
+      return await originalMethod.apply(this, args);
+    } catch (err) {
+      if (err.code && err.errno && err.message) {
```
Comment (vbudhram) [inline packages/fxa-graphql-api/src/gql/lib/error.ts:20]: Kind of stinks that auth-server and profile-server return different error objects and this is the only way to tell them apart.

Is there something in graphql that might make this better?

**Review by vbudhram — APPROVED** [review]
@chenba Should we add some basic tests for this? Otherwise LGTM 👍🏽 

Screen Shot 2021-01-04 at 10 28 35 AM

**Comment by bbangert** [comment]
@chenba In NestJS, the correct way to do this would be to apply an Interceptor or ExceptionFilter on the class. There is a default interceptor which catches the errors and reports them (Sentry), but does not apparently handle reformatting the exceptions the same as the prior gateway exception catch did.

**Review by bbangert — CHANGES_REQUESTED** [review]
See other comment on using a NestJS filter.

**Review by bbangert — APPROVED** [review]
This can merge for now, we can do this with an exception filter in a follow-up.

**Pull request merged**